x/executionlayer/keeper.go

```diff
@@ -195,13 +188,15 @@ func (k ExecutionLayerKeeper) Execute(ctx sdk.Context,
 	}
 
 	// Commit
-	postStateHash, bonds, errGrpc := grpc.Commit(k.client, unitHash.EEState, effects, &protocolVersion)
+	postStateHash, bonds, errGrpc := grpc.Commit(k.client, stateHash, effects, &protocolVersion)
 	if errGrpc != "" {
 		return fmt.Errorf(errGrpc)
 	}
 
-	k.SetEEState(ctx, copiedBlockhash, postStateHash)
-	k.SetCandidateBlockBond(ctx, bonds)
+	candidateBlock := ctx.CandidateBlock()
+	candidateBlock.State = postStateHash
+	candidateBlock.Bonds = bonds
+	ctx = ctx.WithCandidateBlock(candidateBlock)
```

Comment:
Because it treats candidateBlock as a pointer, it is not necessary to call WithCandidateBlock.(3)

Reply:
https://github.com/hdac-io/friday/pull/58/commits/6f1742302db2b5a043fd34ef8f04767522701f87